src/keystore.h

```diff
@@ -145,12 +145,14 @@ class CKeyStore
     virtual std::optional<AddressUFVKMetadata> GetUFVKMetadataForReceiver(
             const libzcash::Receiver& receiver) const = 0;
 
+    std::optional<AddressUFVKMetadata> GetUFVKMetadataForAddress(
+            const CTxDestination& address) const;
```

Comment:
I couldn’t make this an overload of `GetUFVKMetadataForReceiver` because both just take a `typedef` over a `variant`. Happy to change the name if you have a better one.

Reply:
Every name I can come up with here is absurdly long. 

C++ question (though not for this PR) that I don't know the answer to: I've wanted to get rid of `CTxDestination` entirely for a long time; it's not in upstream Bitcoin, it's something that we added. If `CTxDestination` were a std::variant that is a *subset* of the `libzcash::Receiver` type (and got rid of the `NoDestination` variant or whatever it is) would we then be able to use the regular `GetUFVKMetadataForReceiver`?

Reply:
Hello, it's your friendly neighbourhood lore-bearer here! 😅

`CTxDestination` is _not_ something that we added. We [inherited it from Bitcoin Core 0.11.2](https://github.com/bitcoin/bitcoin/blob/7e278929df53e1fb4191bc5ba3176a177ce718bf/src/script/standard.h#L77-L84), and it is [still present in current upstream](https://github.com/bitcoin/bitcoin/blob/23e2bfcbc42849daa8e2b69f9bbdc526bc8743a7/src/script/standard.h#L138-L149). It is a pretty fundamental type throughout the Bitcoin Core wallet infrastructure, which is why I've not attempted to remove it despite my annoyance with it existing separately from our other address types (because doing so would involve changes to almost every single legacy transparent-only wallet method).

If it were used as `std::optional<CTxDestination>` then we could just have `CTxDestination` be a sub-case of our overarching receiver type, but the `NoDestination` variant makes that messy. I presume it exists because `std::optional` was only added in C++17, and `CTxDestination` was introduced in 2012 (https://github.com/bitcoin/bitcoin/pull/1357).

As a side-note, you can usually tell if a class is from us or upstream by whether its name starts with `C`. We never liked using Hungarian Notation for classes, so we (almost) never did for types we introduced. For the few (generally early) exceptions it's easy to distinguish them via `git blame`.